nautobot/extras/models/groups.py

```diff
@@ -0,0 +1,283 @@
+
+            else:
+                new_value = field_value
+                logger.debug("%s: %s", field_name, field_value)
+
+            # Don't store empty values like `None`, [], etc.
+            if new_value in (None, "", [], {}):
+                logger.debug("Not storing empty value (%s) for %s", field_value, field_name)
+                continue
+
+            new_filter[field_name] = new_value
+
+        self.filter = new_filter
+
+    # FIXME(jathan): Yes, this is "something", but there is discrepancy between explicitly declared
+    # fields on `DeviceFilterForm` (for example) vs. the `DeviceFilterSet` filters. For example
+    # `Device.name` becomes a `MultiValueCharFilter` that emits a `MultiValueCharField` which
+    # expects a list of strings as input. The inverse is not true. It's easier to munge this
+    # dictionary when we go to send it to the form, than it is to dynamically coerce the form field
+    # types coming and going... For now.
+    def get_initial(self):
+        """
+        Return an form-friendly version of `self.filter` for initial form data.
+
+        This is intended for use to populate the dynamically-generated filter form created by
+        `generate_filter_form()`.
+        """
+        filter_fields = self.get_filter_fields()
+        initial_data = self.filter.copy()
+
+        # Brute force to capture the names of any `*CharField` fields.
+        char_fields = [f for (f, ftype) in filter_fields.items() if ftype.__class__.__name__.endswith("CharField")]
+
+        # Iterate the char fields and coerce their type to a singular value or
+        # an empty string in the case of an empty list.
+        for char_field in char_fields:
+            field_value = initial_data.get(char_field, "not_found")
+            if field_value == "not_found":
+                continue
+
+            if isinstance(field_value, list):
+                # Either the first (and should be only) item in this list.
+                if field_value:
+                    new_value = field_value[0]
+                # Or empty string if there isn't.
+                else:
+                    new_value = ""
+                initial_data[char_field] = new_value
+
+        return initial_data
+
+    def generate_filter_form(self):
+        """
+        Generate a `FilterForm` class for use in `DynamicGroup` edit view.
+
+        This form is used to popoulate and validate the filter dictionary.
+        """
+        filterform_class = self.map and self.map.filterform_class
+        filter_fields = self.get_filter_fields()
+
+        # FIXME(jathan): Account for field_order in the newly generated class.
+        try:
+
+            class FilterForm(filterform_class):
+                prefix = "filter"
+
+                def __init__(self, *args, **kwargs):
+                    super().__init__(*args, **kwargs)
+                    self.fields = filter_fields
+
+        except (AttributeError, TypeError):
+            return None
```

Comment:
When do we expect to hit this case? Would it be better to just let the exception bubble up to the caller?

Reply:
In the castoff `map`, no, because of underlying `@classproperty` stuff and the entire class-based methodology of `BaseDynamicGroupMap`. I'm going to just rip it all out and move it into the model class as indicated in another comment. There is NO use-case for a map without a model class anyhow.